lib/administrate/field/number.rb

```diff
@@ -1,16 +1,19 @@
 require_relative "base"
+require "active_support/number_helper"
 
 module Administrate
   module Field
     class Number < Field::Base
       def to_s
-        data.nil? ? "-" : format_string % value
+        result = data.nil? ? "-" : format_string % value
+        result = add_delimiter(result) if options[:delimiter]
+        prefix + result + suffix
```

Comment:
I realized that the delimiter option works fine as originally coded, but only when there isn't a prefix or suffix. Commit b163adb ensures it works with both and adds a test to ensure that that's the case.